rcl_action/test/rcl_action/test_action_server.cpp

```diff
@@ -308,6 +389,74 @@ TEST_F(TestActionServer, test_action_accept_new_goal)
   }
 }
 
+TEST_F(TestActionServer, test_action_server_goal_exists) {
+  rcl_action_goal_info_t goal_info_out = rcl_action_get_zero_initialized_goal_info();
+  EXPECT_FALSE(rcl_action_server_goal_exists(nullptr, &goal_info_out));
+  EXPECT_TRUE(rcl_error_is_set());
+  rcl_reset_error();
+
+  EXPECT_FALSE(rcl_action_server_goal_exists(&this->action_server, nullptr));
+  EXPECT_TRUE(rcl_error_is_set());
+  rcl_reset_error();
+
+  // Initialize a goal info
+  rcl_action_goal_info_t goal_info_in = rcl_action_get_zero_initialized_goal_info();
+  init_test_uuid0(goal_info_in.goal_id.uuid);
+
+  // Add new goal
+  rcl_action_goal_handle_t * goal_handle =
+    rcl_action_accept_new_goal(&this->action_server, &goal_info_in);
+  EXPECT_NE(goal_handle, nullptr) << rcl_get_error_string().str;
+
+  // Check exists
+  EXPECT_TRUE(rcl_action_server_goal_exists(&this->action_server, &goal_info_in));
+
+  rcl_action_goal_info_t different_goal = rcl_action_get_zero_initialized_goal_info();
+  init_test_uuid1(goal_info_in.goal_id.uuid);
+
+  // Check doesn't exist
+  EXPECT_FALSE(rcl_action_server_goal_exists(&this->action_server, &different_goal));
```

Comment:
If we're checking `rcl_error_is_set()` in the failure cases, I suppose it also makes sense to check in the nominal cases. E.g.

```c++
EXPECT_FALSE(rcl_error_is_set());
```

Reply:
Done.